app/models/deleted_object.rb

```diff
@@ -8,5 +8,14 @@ class DeletedObject < ApplicationRecord
     scope scoped_class.to_s.underscore.pluralize.to_sym, -> { where(object_type: scoped_class) }
   end
 
-  # TODO: cleanup regularly once the owner is destroyed too :) or this will grow uncontrollably
+  scope(:deleted_owner, lambda do
+    join_sql = <<-SQL
+      INNER JOIN deleted_objects deleted_owner
+        ON deleted_objects.owner_type = deleted_owner.object_type
+        AND deleted_objects.owner_id = deleted_owner.object_id
+    SQL
+    joins(join_sql)
+  end)
+
+  scope :stale, -> { where.has { ((id.in DeletedObject.deleted_owner.select(:id)) | (object_type == Service.name)) & (created_at <= 1.week.ago) } }
```

Comment:
what's the meaning of `|` in this query?

Reply:
@macejmic It is an 'OR' in [baby_squeel](https://github.com/rzane/baby_squeel)